### PR TITLE
Makefile adaptations for Py3/2 and Jenkins configuration

### PR DESCRIPTION
**Pull request by luizirber**
- Jenkins: Reuse PYTHON_EXECUTABLE from the environment.
- Generate the appropriate name for the Python extension module based on the correct version and implementation.
- Update coverage arguments.

### CODE REVIEW

**Comment by mr-c** [comment]
Jenkins, test this please

**Comment by mr-c** [comment]
Turns out the coverage levels have to be adjusted lower for Py3.x (and are different for py3.3 vs 3.4). We should merge this soon

**Comment by luizirber** [comment]
Maybe we should combine the coverage? Something like [this][1]? And don't set as failing on each job, but just on the aggregate?

[1]: https://github.com/dib-lab/khmer/commit/d6ba5d6f666ff218374dd51ee4b149caba752211#diff-b91f3d5bd63fcd17221b267e851608e8R8

**Comment by mr-c** [comment]
Combining coverage would be nifty. I guess we could publish the coverage
files as artifacts?

I can't find how to configure the success criteria for the parent job

On Fri, Jun 19, 2015 at 7:06 PM Luiz Irber <notifications@github.com> wrote:

> Maybe we should combine the coverage? Something like this
> <https://github.com/dib-lab/khmer/commit/d6ba5d6f666ff218374dd51ee4b149caba752211#diff-b91f3d5bd63fcd17221b267e851608e8R8>?
> And don't set as failing on each job, but just on the aggregate?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1108#issuecomment-113695959>.
>
-- 
Michael R. Crusoe: Programmer & Bioinformatician crusoe@ucdavis.edu
<mcrusoe@msu.edu>
The lab for Data Intensive Biology; University of California, Davis
https://impactstory.org/MichaelRCrusoe http://twitter.com/biocrusoe

**Comment by mr-c** [comment]
Finally this builds cleanly.

File: Makefile [hunk 1]
```diff
@@ -157,7 +157,8 @@ diff_pylint_report: pylint_report.txt
 # statement). So we run nose inside of coverage.
 .coverage: $(PYSOURCES) $(wildcard tests/*.py) khmer/_khmermodule.so
 	coverage run --branch --source=scripts,khmer,oxli --omit=khmer/_version.py \
-		-m nose --with-xunit --attr=\!known_failing --processes=0
+		-m nose --with-xunit --attr=\!known_failing,\!huge \
```
Comment (kdmurray91) [inline Makefile]: As of master on my laptop, you can also just use `--attr $(TEST_ATTR)` so that the attrs are the same everywhere

Comment (kdmurray91) [inline Makefile]: And use `TEST_ATTR ?= XXX` up the top of the makefile, and `export TEST_ATTR='!known_failing,!huge'` in jenkins.sh

Comment (kdmurray91) [inline Makefile]: P.S. I have a patch for these, if you'd like me to push it @luizirber

Comment (luizirber) [inline Makefile]: Go for it! I don't know what are our guidelines for cases like this, but if you don't want to commit directly to this branch you can always make a PR for other PRs. But I don't mind if you do directly =]

**Comment by luizirber** [comment]
- [x] Is it mergeable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Is the Copyright year up to date?

Ready for review and merge @mr-c @kdmurray91

**Comment by luizirber** [comment]
Made this mergeable again, ready to review @mr-c 

(the build is broken because #1176 was merged without changing `runscriptredirect` to `run_shell_cmd`, I fixed it in #1204 )

**Comment by mr-c** [comment]
LGTM to me! Will merge after build is un-broken

**Comment by mr-c** [comment]
Jenkins, retest this please

**Pull request merged**